app/models/task.rb

```diff
@@ -264,6 +264,34 @@ def do_hold(last_check)
     end
   end
 
+  def name
```

Comment:
`#type_name_display` is only used for the _task_list.html.erb, so it can be modified. However, 
1) It uses #titlecase, so it should already be adding spaces between words and capitalizing appropriately which handles your first problem there. You can test in the Rails Console:
```bash
2.2.1 :001 > a = "TurnOnTask"
 => "TurnOnTask" 
2.2.1 :002 > a.titlecase
 => "Turn On Task" 

```
2) Each RIMS has a Thermostat component and can be tasked with a RunThermostatTask. As such, while it's a little janky to have "Run Thermostat Thermostat 4", I think it's still appropriate. If we really hate it in the long run it's easier to customize it in the one spot than duplicate the function.

So, I'd say use `#type_name_display` :)